crates/forge/src/lib.rs

```diff
@@ -462,35 +488,101 @@ fn run_with_fuzzing(
     };
     let mut fuzzer = RandomFuzzer::create(fuzzer_seed, fuzzer_runs, &args)?;
 
-    let mut results = vec![];
+    let mut tasks = FuturesUnordered::new();
+
+    let (send, mut recv) = channel(1);
 
     for _ in 1..=fuzzer_runs {
         let args = fuzzer.next_args();
+        let case = case.clone();
+        let case_copy = case.clone();
+        let runner = runner.clone();
+        let args = args.clone();
+        let cancellation_fuzzing_token = token.clone();
+        let runner_config = runner_config.clone();
+        let send = send.clone();
+
+        tasks.push(task::spawn(async move {
+            tokio::select! {
+                () = runner_config.error_cancellation_token.cancelled() => {
+                    //Stop executing all tests because
+                    //one of a test returns Err
```

Comment:
```suggestion
                    //Stop executing all tests because
                    //one of a test returns Err
```
```suggestion
                    // Stop executing all tests because
                    // one of a test returns Err
```